src/tasks/processBulkUpload.ts

```diff
@@ -0,0 +1,18 @@
+import { getLogger } from '../logger';
+import { isProcessBulkUploadJobPayload } from '../types';
+import { InternalValidationError } from '../errors';
+
+const logger = getLogger(__filename);
+
+export const processBulkUpload = async (
+  payload: unknown,
+): Promise<void> => {
+  if (!isProcessBulkUploadJobPayload(payload)) {
+    logger.debug(isProcessBulkUploadJobPayload.errors, 'Malformed bulk upload job payload');
+    throw new InternalValidationError(
```

Comment:
This raises a question I hadn't really considered before: how do we mark bulk uploads as failed? From the [error handling docs](https://worker.graphile.org/docs/error-handling),

> If a job throws an error, the job is failed and scheduled for retries with exponential back-off.

It's great that the framework will handle that kind of failure for us, but it's not obvious to me how/when/where to handle the follow-up we need to do of connecting a job error to a bulk upload status.

I think we can probably defer that out of this PR, since it's just setting up the framework, but if you've already thought about it I'd love to hear where you're at!

Reply:
@slifty and I talked about this on a call today. It's definitely out of scope for this PR, but broadly the worker code will update the bulk upload status as it goes. For now, monitoring job failures will be a manual process, which we think is the appropriate engineering trade-off for the moment given our (lack of) scale.